src/components/comments-not-connected/comments-not-connected.js

```diff
@@ -0,0 +1,112 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+
+import {
+  GetDisqusUser,
+  ConnectUserCommentsAccount,
+  DisqusError,
+} from '../../services/apiService';
+import DisqusIcon from '-!svg-react-loader!../../images/disqusIcon.svg';
+
+import { useAuth0 } from '@auth0/auth0-react';
+import { ApplicationInsights } from '@microsoft/applicationinsights-web';
+const appInsights = new ApplicationInsights({
+  config: {
+    instrumentationKey: process.env.APPINSIGHTS_INSTRUMENTATIONKEY,
+  },
+});
+
+const CommentsNotConnected = ({
+  userCommentsConnected,
+  setListChange,
+  listChange,
+}) => {
+  const [disqusUsername, setDisqusUsername] = useState();
+  const [errorMessage, setErrorMessage] = useState(null);
+
+  const { user, getIdTokenClaims } = useAuth0();
+
+  useEffect(() => {}, [userCommentsConnected, listChange]);
+  return (
+    <div className="connect-acc-container">
+      <DisqusIcon className="disqus-large-icon" />
+      <div className="form">
+        <div>
+          <input
+            className={
+              !errorMessage ? 'username-input-box' : 'username-input-box-error'
+            }
+            type="text"
+            name="disqusId"
+            value={disqusUsername}
+            placeholder="Enter Disqus Username"
+            onChange={(e) => setDisqusUsername(e.target.value)}
+          />
+          <div className="forgot-username-tooltip">
+            {/* eslint-disable-next-line jsx-a11y/anchor-has-content */}
+            <a
+              href="https://disqus.com/home/"
+              target="_blank"
+              rel="noreferrer"
+              className="get-username-btn unstyled"
+            />
+            <span className="tooltiptext">Forgot username? </span>
+          </div>
+          <p className="error-text">{errorMessage}</p>
+        </div>
+
+        <button
+          className="connect-acc-button"
+          onClick={() => {
+            if (disqusUsername) {
+              GetDisqusUser(disqusUsername)
```

Comment:
I would always put javascript logic into a separate area so it is clearly separated from the jsx/html code, but you could argue this is a stylistic choice. What do you think?

Reply:
To clarify I mean making this into a function.

Reply:
As per our conversation, I agree that the logic should be seperate from the html. I made this change in this commit https://github.com/SSWConsulting/SSW.Rules/pull/468/commits/352601d604a11c40b8859d88e883bc6eb32d2d75